firmware/defmt-rtt/build.rs

```diff
@@ -0,0 +1,23 @@
+fn main() {
+    println!("cargo:rerun-if-env-changed=DEFMT_RTT_BUFFER_SIZE");
+
+    let size = option_env!("DEFMT_RTT_BUFFER_SIZE")
+        .map(|s| s.parse().ok())
+        .flatten()
+        .unwrap_or(1024_usize);
+
+    let non_power_of_two = (size & (size - 1)) != 0;
+
+    if non_power_of_two {
+        println!("cargo:warning=RTT buffer size of {} is not a power of two, performance will be degraded.", size)
+    }
```

Comment:
Since this won't show up for most users, and has no way to silence it if it does, maybe remove it entirely?

Reply:
I'm unfortunately not following, can you describe a concrete problem case?

Reply:
As an aside, `size.count_ones() == 1` should do the same thing. But yeah, warnings don't help here. The other option is having two env vars, one for `RTT_BUFFER_SIZE` and one for `RTT_BUFFER_SIZE_NON_ALIGNED` and panic if the first isn't 2**N.

Reply:
I was under the impression that a user would configure this variable to conserve memory (at least, that's my personal use case and why I opened this issue/can of worms in the first place); what am I missing re: warnings don't help?

Reply:
Warnings don't show up for users unless they use defmt-rtt as a path dependency